### DISCUSSION TITLE
sqlite jsonb support

### DISCUSSION THREAD

**Discussion by anacrolix** [duckdb/duckdb#13432] [Ideas]
sqlite has a more efficient json storage format, JSONB. It doesn't seem to be supported. Without support for this it's necessary to duplicate JSONB columns into textual JSON so duckdb can use them.

Please let me know if this belongs somewhere else, and if there's another location where sqlite specific support is described/discussed. Thanks

**Comment by rustyconover** [comment]
I've been looking into adding JSONB support via a Rust crate, it doesn't seem too difficult.

**Reply by anacrolix** [reply]
I'm on my phone or I'd do more checks but is this an issue?

> The "JSONB" name is inspired by PostgreSQL, but the on-disk format for SQLite's JSONB is not the same as PostgreSQL's. The two formats have the same name, but they have wildly different internal representations and are not in any way binary compatible.

**Reply by anacrolix** [reply]
Nevermind I followed up and found https://github.com/lovasoa/serde-sqlite-jsonb.
